### PR TITLE
Fixes SI-6354: improved error messages for Dynamic signature mismatches.

### PR DESCRIPTION
**Pull request by namin**
If an error occurs afer a Dynamic rewriting, augment the error message
with the rewritten tree and a hint to check the Dynamic method
signature.

Review by @hubertp.

### CODE REVIEW

File: test/files/neg/applydynamic_sip.check [hunk 1]
```diff
@@ -7,4 +7,52 @@ applydynamic_sip.scala:8: error: applyDynamicNamed does not support passing a va
 applydynamic_sip.scala:9: error: applyDynamicNamed does not support passing a vararg parameter
   qual.sel(arg, arg2 = "a2", a2: _*)
        ^
-three errors found
+applydynamic_sip.scala:18: error: type mismatch;
+ found   : String("sel")
+ required: Int
+possible cause: maybe a wrong Dynamic method signature?
+rewrite to Test.this.bad1.selectDynamic("sel")
```
Comment (xeno-by) [inline test/files/neg/applydynamic_sip.check]: Does this suggest the programmer to rewrite the offending code to `Test.this.bad1.selectDynamic("sel")` to get rid of the error?

Or does this mean that the offending code has been rewritten, and this produced an error?

Comment (gkossakowski) [inline test/files/neg/applydynamic_sip.check]: I agree with @xeno-by that the wording should be changed.

Comment (namin) [inline test/files/neg/applydynamic_sip.check]: @xeno-by, how about:

after rewriting to Test.this.bad1.selectDynamic("sel")
possible cause: maybe a wrong Dynamic method signature?

Or any suggestions?

**Comment by hubertp** [comment]
lgtm

**Pull request merged**